enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/KeyValidator.java

```diff
@@ -149,13 +151,26 @@ public static APIKeyValidationInfoDTO validateSubscription(APIConfig apiConfig,
                 app = datastore.getApplicationById(key.getApplicationUUID());
                 if (app != null) {
                     sub = datastore.getSubscriptionById(app.getUUID(), uuid);
-                    if (sub != null) {
-                        log.debug("All information is retrieved from the inmemory data store.");
+                    if (sub == null) {
+                        try {
+                            SemVersion apiSemVersion = SemanticVersionUtil.validateAndGetVersionComponents(apiVersion,
+                                    uuid);
+                            String apiContextTemplate = apiContext.replace("/" + apiVersion, "");
```

Comment:
Since there can be user who use context like /foo/1.0.0/1.0.0 by accident, better to have a replace last to avoid that.